src/server/pfs/server/driver.go

```diff
@@ -1830,22 +1834,22 @@ func (d *driver) filePathFromEtcdPath(etcdPath string) string {
 	return path.Join(split[4:]...)
 }
 
-func (d *driver) putFile(ctx context.Context, file *pfs.File, delimiter pfs.Delimiter,
-	targetFileDatums int64, targetFileBytes int64, overwriteIndex *pfs.OverwriteIndex, reader io.Reader) error {
-	pachClient := d.getPachClient(ctx)
-	ctx = pachClient.Ctx()
-	if err := d.checkIsAuthorized(pachClient, file.Commit.Repo, auth.Scope_WRITER); err != nil {
+func (d *driver) putFiles(s *putFileServer) error {
+	ctx := s.Context()
+	req, err := s.Peek()
```

Comment:
So what is the reason for the peek here? Is it to catch potential errors earlier?